### PR TITLE
Prompt user on empty validator.db

### LINKED ISSUE #7331: Cannot initialize validator.db: permission denied
# 🐞 Bug Report

### Description

Hi, I can't upgrade my validators as version `v1.0.0-alpha.25` introduced a breaking change of moving `validator.db` from `data` directory to the `wallet` directory. In the Kubernetes environment where these two directories are separated (one is mounted as `rw` and is PV and another is mounted as `ro` and is Kubernetes secret), there is no way to combine those two under the same path.

Please provide a way to separate `validator.db` and `wallet` directories.

### Has this worked before in a previous version?

Yes, it has stopped working after introducing the following change: https://github.com/prysmaticlabs/prysm/pull/7176

### PR DESCRIPTION
**Pull request by shayzluf**
**What type of PR is this?**

Bug fix

why?
two reasons:

extra protection layer
missing db file is a slashing protection problem. warn users and allow them to copy the db file if its empty
allowing users to define their custom protection db location
some users reported they don't have write permissions to wallet dir and need to configure a different location for it
What does this PR do? Why is it needed?

use wallet dir flag as default protection db dir
if datadir flag is set use it for protection db (we introduced a breaking change)
if file doesn't exist warn user and ask him to give the path to the latest copy
Copy existing db from old path to datadir
Which issues(s) does this PR fix?

Fixes #7331
Screen Shot 2020-10-23 at 8 54 42

### CODE REVIEW

File: validator/node/node.go [hunk 1]
```diff
@@ -202,13 +201,14 @@ func (s *ValidatorClient) initializeFromCLI(cliCtx *cli.Context) error {
 			log.Fatalf("Could not get a lock on wallet file. Please check if you have another validator instance running and using the same wallet: %v", err)
 		}
 	}
-
-	dataFlag := flags.WalletDirFlag
+	dataDir := cliCtx.String(flags.WalletDirFlag.Name)
```
Comment (nisdas) [inline validator/node/node.go:204]: So this location will never have a db, it is not the root of the accounts directory. Just specifies the general folder to keep
all wallets in prysm, using it as the default in the event an account is not initialized will lead to undefined behavior.

Comment (shayzluf) [reply validator/node/node.go:204]: https://github.com/prysmaticlabs/prysm/blob/ab76bdad150a5c38e394b213ecf584c30cf887ac/validator/node/node.go#L175 wallet is not defined here

Comment (shayzluf) [reply validator/node/node.go:204]: wallet is not "" https://github.com/prysmaticlabs/prysm/blob/ab76bdad150a5c38e394b213ecf584c30cf887ac/validator/flags/flags.go#L130

Comment (shayzluf) [reply validator/node/node.go:204]: @nisdas wdyt?

**Comment by rauljordan** [comment]
@shayzluf I have been thinking about this feature a lot more, and tried it a few times on a local setup, trying to envision myself from the perspective of a staker and here are some honest thoughts that I didn't foresee upon first discussing the PR:

## The Problem We're Trying to Solve

As I understand, we notice that some users have different permissions in their slashing protection db, which was previously stored in the datadir, and this prevents us from manipulating that file now. This issue has not been reported by many users, only several that have unfortunately suffered some consequences such as https://github.com/prysmaticlabs/prysm/issues/7331.

## Issues

The problem with this solution is how it causes more harm than good due to being confusing in nature, only serving a few edge cases where it is not worth it for us to fix at this point. Trying out the feature myself, it was indeed terrifying to see the log pop-up upon starting my node. I believe we'll get a lot of very upset users thanks to this feature.

1. The motivation is not clear to stakers at all. The messaging is confusing:

```
Warning! The protection DB is essential to prevent slashing. If the current " +
	"validator keys have been used before, \nplease specify the location of the DB file."
var defaultWarning = "By default, the DB file is named validator.db . Locate this file from " +
	"your previous validator setup, and paste its \npath below. It will be copied to the existing configurations directory.
```

2. Most stakers have zero knowledge of how slashing protection internals work. This message scares me because (1) I had no idea I needed a validator.db in the first place and makes me wonder if I should have known, (2) it is not clear why we are doing this, what problem is it solving, did I mess something up previously?

3. We are already creating new legacy code. Having business logic for migrations is already adding more code bloat that we'll have to delete one day to support a latest version (Specifically, having code for the migration)

4. It is a breaking change that we have a chance to introduce in beta, but has potential to go wrong and cause more harm than good if there is a bug

## Path Forward

I understand this issue is a problem https://github.com/prysmaticlabs/prysm/issues/7331, but perhaps it is best to simply let it remain. The trade-offs are too much just to support legacy code that will support a few Medalla users that had such custom setups. Let's revert these features, leave them as is, and do not cause such a disruptive, breaking change.

**Comment by shayzluf** [comment]
@nisdas would we like to reconsider?

**Comment by shayzluf** [comment]
closed i favour of a less confusing user flow #7791

**Pull request closed without merging**